resim/metrics/default_report_metrics.py

```diff
@@ -0,0 +1,729 @@
+        project_id: The id for the project for this job.
+        batch_ids: THe ids for all the batches we need metrics for.
+
+    Returns:
+        A dictionary of batch_ids to batch metrics for all of our batches.
+    """
+    return await dict_gather(
+        {
+            batch_id: _fetch_batch_metrics_for_batch(
+                client, project_id=str(project_id), batch_id=batch_id
+            )
+            for batch_id in batch_ids
+        }
+    )
+
+
+async def fetch_scalar_batch_metrics(
+    *, batch_to_metrics_map: dict[str, list[BatchMetric]]
+) -> dict[str, dict[str, MetricProto]]:
+    """Fetch all scalar metric protos for all of our batches.
+
+    Scalar metrics for each batch are very useful to display in reports and we
+    need to fetch and unpack the protos to get the units from them. This
+    function does this.
+
+    Args:
+        batch_to_metrics_map: A map containing all of the batch metrics for each
+                              batch. Keyed by batch id.
+    Returns:
+        A dictionary of batch_id to dict of batch metric name to metric proto
+        object.
+    """
+    client = httpx.AsyncClient()
+    tasks = {}
+
+    async def unpack(query: Awaitable) -> MetricProto:
+        response = await query
+        message = MetricProto()
+        message.ParseFromString(response.content)
+        return message
+
+    for batch, metrics in batch_to_metrics_map.items():
+        batch_tasks = {}
+        for metric in metrics:
+            if metric.type == MetricType.SCALAR:
+                batch_tasks[metric.name] = unpack(client.get(metric.metric_url))
+        tasks[batch] = dict_gather(batch_tasks)
+    responses = await dict_gather(tasks)
+    return responses
+
+
+def _count_job_statuses(jobs: list[Job]) -> list[int]:
+    """Count the different output statuses in a given job list.
+
+    Count job statuses into the following categories in order:
+
+    PASSED
+    FAIL_WARN
+    FAIL_BLOCK
+    ERROR
+    CANCELLED
+    UNKNOWN
+
+    Args:
+        jobs: The jobs whose statuses we care about.
+
+    Returns:
+        A list of counts in the aforementioned order.
+    """
+    counts: dict[JobStatus, MetricStatus] = defaultdict(int)
+    for job in jobs:
+        counts[(job.job_status, job.job_metrics_status)] += 1
+
+    passed_count = counts[(JobStatus.SUCCEEDED, MetricStatus.PASSED)]
+    fail_warn_count = counts[(JobStatus.SUCCEEDED, MetricStatus.FAIL_WARN)]
+    fail_block_count = counts[(JobStatus.SUCCEEDED, MetricStatus.FAIL_BLOCK)]
+    error_count = sum(counts[JobStatus.ERROR, ms] for ms in MetricStatus)
+    cancelled_count = sum(counts[JobStatus.CANCELLED, ms] for ms in MetricStatus)
+
+    categorized_counts = [
+        passed_count,
+        fail_warn_count,
+        fail_block_count,
+        error_count,
+        cancelled_count,
+    ]
+    # Add unknown column
+    categorized_counts.append(len(jobs) - sum(categorized_counts))
```

Comment:
yep, makes sense: ignore i believe the worker explicitly sets any running job to metrics status unknown anyway, so this is the right logic